webpack/assets/javascripts/react_app/components/Layout/Layout.js

```diff
@@ -5,146 +5,88 @@ import { VerticalNav } from 'patternfly-react';
 
-class Layout extends React.Component {
-  componentDidMount() {
-    const {
-      items,
-      data,
-      fetchMenuItems,
-      changeLocation,
-      currentLocation,
-      changeOrganization,
-      currentOrganization,
-      changeActiveMenu,
-      activeMenu,
-      isCollapsed,
-      onCollapse,
-    } = this.props;
-    if (items.length === 0) fetchMenuItems(data);
-    if (isCollapsed) onCollapse();
-
-    const activeURLMenu = getActive(data.menu, getCurrentPath());
-    if (activeMenu !== activeURLMenu.title) {
-      changeActiveMenu(activeURLMenu);
-    }
-
-    if (
-      data.taxonomies.locations &&
-      !!data.locations.current_location &&
-      currentLocation !== data.locations.current_location
-    ) {
-      const initialLocTitle = data.locations.current_location;
-      const initialLocId = data.locations.available_locations.find(
-        loc => loc.title === initialLocTitle
-      ).id;
-      changeLocation({ title: initialLocTitle, id: initialLocId });
-    }
-
-    if (
-      data.taxonomies.organizations &&
-      !!data.orgs.current_org &&
-      currentOrganization !== data.orgs.current_org
-    ) {
-      const initialOrgTitle = data.orgs.current_org;
-      const initialOrgId = data.orgs.available_organizations.find(
-        org => org.title === initialOrgTitle
-      ).id;
-      changeOrganization({ title: initialOrgTitle, id: initialOrgId });
-    }
-  }
-
-  render() {
-    const {
-      items,
-      data,
-      isLoading,
-      isCollapsed,
-      onExpand,
-      onCollapse,
-      changeActiveMenu,
-      changeOrganization,
-      changeLocation,
-      currentOrganization,
-      currentLocation,
-      activeMenu,
-      children,
-      history,
-    } = this.props;
-
-    return (
-      <React.Fragment>
-        <VerticalNav
-          hoverDelay={100}
-          items={items}
-          onItemClick={primary =>
-            handleMenuClick(primary, activeMenu, changeActiveMenu)
+const Layout = ({
+  items,
+  data,
+  isLoading,
+  isCollapsed,
+  navigate,
+  expandLayoutMenus,
+  collapseLayoutMenus,
+  changeActiveMenu,
+  changeOrganization,
+  changeLocation,
+  currentOrganization,
+  currentLocation,
+  activeMenu,
+  children,
+}) => (
+  <React.Fragment>
+    <VerticalNav
+      hoverDelay={100}
+      items={items}
+      onItemClick={primary =>
+        handleMenuClick(primary, activeMenu, changeActiveMenu)
+      }
+      onNavigate={({ href }) => navigate(href)}
+      activePath={`/${__(activeMenu || 'active')}/`}
+      onCollapse={collapseLayoutMenus}
+      onExpand={expandLayoutMenus}
+    >
+      <VerticalNav.Masthead>
+        <VerticalNav.Brand
+          title={data.brand}
+          iconImg={data.logo}
+          href={data.root}
+        />
+        <TaxonomySwitcher
+          taxonomiesBool={data.taxonomies}
+          currentLocation={currentLocation}
+          locations={
+            data.taxonomies.locations ? data.locations.available_locations : []
```

Comment:
one nit: since you are already improving structure, you can also use one of the [helpers](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/common/helpers.js#L112) to transform object keys from snake case to camel case to get a consistent style. There is also a [variant](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/common/helpers.js#L127) that does that kind of transform in depth.

Reply:
It turned to be too complicated because those props go deep to other components and also used by the ReactApp itself so I think it is better to do that in a different PR